server/autolinkclient/client.go

```diff
@@ -45,22 +49,79 @@ func (c *Client) Add(links ...autolink.Autolink) error {
 			return err
 		}
 
-		req, err := http.NewRequest("POST", "/"+autolinkPluginID+"/api/v1/link", bytes.NewReader(linkBytes))
+		resp, err := c.call("/"+autolinkPluginID+"/api/v1/link", http.MethodPost, linkBytes, nil)
 		if err != nil {
 			return err
 		}
 
-		resp, err := c.Do(req)
+		if resp.StatusCode != http.StatusOK {
+			respBody, _ := ioutil.ReadAll(resp.Body)
+			return fmt.Errorf("unable to add the link %s. Error: %v, %v", link.Name, resp.StatusCode, string(respBody))
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) Delete(links ...string) error {
+	for _, link := range links {
+		queryParams := url.Values{
+			AutolinkNameQueryParam: {link},
+		}
+
+		resp, err := c.call("/"+autolinkPluginID+"/api/v1/link", http.MethodDelete, nil, queryParams)
 		if err != nil {
 			return err
 		}
-		defer resp.Body.Close()
 
 		if resp.StatusCode != http.StatusOK {
 			respBody, _ := ioutil.ReadAll(resp.Body)
-			return fmt.Errorf("unable to install autolink. Error: %v, %v", resp.StatusCode, string(respBody))
+			return fmt.Errorf("unable to delete the link %s. Error: %v, %v", link, resp.StatusCode, string(respBody))
 		}
 	}
 
 	return nil
 }
+
+func (c *Client) Get(autolinkName string) ([]*autolink.Autolink, error) {
+	queryParams := url.Values{
+		AutolinkNameQueryParam: {autolinkName},
+	}
+
+	resp, err := c.call("/"+autolinkPluginID+"/api/v1/link", http.MethodGet, nil, queryParams)
+	if err != nil {
+		return nil, err
+	}
+
+	var respBody []byte
+	if resp.StatusCode != http.StatusOK {
+		respBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("unable to get the link %s. Error: %v, %v", autolinkName, resp.StatusCode, string(respBody))
+	}
```

Comment:
Apologies for my previous comment, I didn't realise that we were using `respBody` in both success and error cases. It's fine to keep reading the `resp` before performing the status code check:

```suggestion
	respBody, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	if resp.StatusCode != http.StatusOK {
		return nil, fmt.Errorf("unable to get the link %s. Error: %v, %v", autolinkName, resp.StatusCode, string(respBody))
	}

```